metrics/prometheus.go

```diff
@@ -33,7 +35,9 @@ func (p *PromMetrics) Start() error {
 		return err
 	}
 
+	p.lock.Lock()
 	p.metrics = make(map[string]interface{})
+	p.lock.Unlock()
```

Comment:
`Start` should only be called once so we shouldn't need this here unless something has gone terribly wrong!

Reply:
I feel like it's worth keeping because it makes it clear that the mutex is protecting writes (but not reads) to the map and doesn't come at any real cost. Whereas debugging any runtime panic does come at a cost. I can remove it if you feel strongly though?

Reply:
I think it suggests to the reader that this method might be called multiple times which would be unexpected and shouldn't happen

Reply:
I would have thought that if it's possible then it's safer to guard it. Your call though, I've rebased to omit it.